arquillian-junit5/src/main/java/org/wildfly/guide/testing/resources/TaskResource.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.guide.testing.resources;
+
+import java.time.Instant;
+import java.util.Collection;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.PATCH;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+
+import org.wildfly.guide.testing.data.TaskRegistry;
+import org.wildfly.guide.testing.model.Task;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@Path("task")
+@RequestScoped
+public class TaskResource {
+
+    @Inject
+    private TaskRegistry taskRegistry;
+    @Inject
+    private UriInfo uriInfo;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Collection<Task> allTests() {
+        return taskRegistry.getTasks();
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/completed/{completed}")
+    public Collection<Task> getAllCompleted(@PathParam("completed") final boolean completed) {
```

Comment:
Out of curiosity, why do the query methods not return a Response object and instead return Lists?

Reply:
It just makes the endpoints look a little cleaner IMO. It also helps if you want some kind of API interface which can work as both a REST endpoint that gets implemented and an REST client, like an MP REST Client.

Either way really works. It's really just a preference and the Jakarta REST implementation handles creating the response for you.